cfme/cloud/flavor.py

```diff
@@ -108,11 +143,65 @@ class Flavor(BaseEntity, Taggable):
     name = attr.ib()
     provider = attr.ib()
 
+    def delete(self, wait=True, cancel=False):
+        """Delete current falvor """
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Remove Flavor', handle_alert=not cancel)
+        view.flash.assert_success_message(
+            'Delete of Flavor "{}" was successfully initiated.'.format(self.name))
+
+        if wait:
+            """Wait for flavor to disappear"""
+            wait_for(lambda: not self.exists,
+                     timeout=600,
+                     message='Wait for flavor to disappear',
+                     delay=20,
+                     fail_func=self.refresh)
+
+    def refresh(self):
+        """Refresh provider relationships and browser"""
+        self.provider.refresh_provider_relationships()
+        self.browser.refresh()
+
+    @property
+    def exists(self):
+        try:
+            navigate_to(self, 'Details')
+            return True
+        except FlavorNotFound:
+            return False
+
 
 @attr.s
 class FlavorCollection(BaseCollection):
     ENTITY = Flavor
 
+    def create(self, name, provider, ram, vcpus, disk, swap, rxtx, is_public=True, cancel=False):
+        view = navigate_to(self, 'Add')
+        form_params = {'provider': provider.name,
+                       'flavor_name': name,
+                       'ram_size': ram,
+                       'vcpus': vcpus,
+                       'disk_size': disk,
+                       'swap_size': swap,
+                       'rxtx_factor': rxtx,
+                       'public': is_public}
+
+        view.fill(form_params)
+
+        if cancel:
+            view.cancel.click()
+            flash_message = "Add of Flavor cancelled by user."
+            view.flash.assert_message(flash_message)
+        else:
+            view.add.click()
+            flash_message = 'Add of Flavor "{}" was successfully initialized.'.format(name)
+            view.flash.assert_success_message(flash_message)
```

Comment:
You almost got it! But starting at line 199, it should not be in the else right? In the if/else block, it is controlling which action button you are clicking and setting the appropriate ```flash_message``` variable for both the cancel and add cases. Both cases will direct you to the parent view, in this case the FlavorAllView. 

The reason we want to create a new view here is because the previous view was set to the FlavorAddView. After a click, either cancel or add, the view is now on the parent view (FlavorAllView). 

Here is an example I just ran through:
```In[3]: collection = appliance.collections.cloud_flavors
In[4]: view = navigate_to(collection, 'Add')
In[5]: view
Out[5]: <cfme.cloud.flavor.FlavorAddView at 0x7f3fa9760e50>
In[6]: view.cancel.click()
In[7]: view
Out[7]: <cfme.cloud.flavor.FlavorAddView at 0x7f3fa9760e50>
```
So when you are checking the flash message, you want to check it on the parent view. 

Also, another thing I just realized. When you cancel adding a Flavor, it is not a success message. You will want to use ```view.flash.assert_message()```. You can use that for both add and cancel cases